vite.config.ts

```diff
@@ -1,53 +1,61 @@
-import { defineConfig, UserConfigExport } from 'vite'
+import { defineConfig, BuildOptions } from 'vite'
 import electron from 'vite-plugin-electron'
 import react from '@vitejs/plugin-react-swc'
 import svgr from 'vite-plugin-svgr'
-import path from 'path'
+import * as path from 'path'
 
 const srcAliases = ['backend', 'frontend', 'common'].map((srcFolder) => {
   return {
     find: srcFolder,
     replacement: path.resolve(__dirname, `./src/${srcFolder}`)
   }
 })
+srcAliases.push({
+  find: '~@fontsource',
+  replacement: path.resolve(__dirname, 'node_modules/@fontsource')
+})
 
-const electronViteConfig: UserConfigExport = {
-  build: { outDir: 'build/electron', target: 'esnext' },
-  resolve: {
-    alias: [
-      {
-        find: '~@fontsource',
-        replacement: path.resolve(__dirname, 'node_modules/@fontsource')
-      },
-      ...srcAliases
-    ]
-  }
+const otherBuildOptions: BuildOptions = {
+  reportCompressedSize: false,
+  target: 'esnext'
 }
 
 export default defineConfig({
   build: {
-    target: 'esnext',
-    outDir: 'build'
+    outDir: 'build',
+    ...otherBuildOptions
   },
   resolve: {
-    alias: [
-      {
-        find: '~@fontsource',
-        replacement: path.resolve(__dirname, 'node_modules/@fontsource')
-      },
-      ...srcAliases
-    ]
+    alias: srcAliases
   },
   plugins: [
     react(),
     electron([
       {
         entry: 'src/backend/main.ts',
-        vite: electronViteConfig
+        vite: {
+          build: {
+            outDir: 'build/main',
```

Comment:
We have to separate the `main` and `preload` scripts now since `vite-plugin-electron` assumes that a script has finished building once the folder it should be in is created (at least it seems like that, haven't checked the code).

If we don't build to separate directories, it builds one script, then thinks both are done and starts Electron, which promptly fails